tooling: Benchmark visualization script

bobluppes opened the pull request:
This PR adds a python script under `tools/`, which can be used to visualize google benchmark data. An example plot can be found below.

Not very familiar with python, so any improvement suggestions are more than welcome.

![vertex](https://github.com/bobluppes/graaf/assets/18380160/e617fb70-cc80-451c-9750-6b264d4e52cd)

joweich commented on tools/benchmark_visualization.py:
```diff
@@ -0,0 +1,57 @@
+
+build/perf/Graaf_perf --benchmark_out=benchmark_results.csv --benchmark_out_format=csv
+"""
+
```
It's an idiomatic [best practice](https://stackoverflow.com/a/419185) to wrap your executed code into `if __name__ == '__main__' ` to avoid unintended side effects. In you're case, it won't make a difference, but it's a good practice nevertheless

joweich reviewed (commented):
@bobluppes Bear with me, I've been very picky with my comments. In general, I think there are some low hanging opportunities to make the code more python-idiomatic. Happy to give it a shot and add my proposal to this PR!

joweich reviewed (approved):
There's definitely still some potential for additional checks and capabilities, esp. automatic inference of `skiprows`, but I think it's fine for now.

bobluppes commented:
> There's definitely still some potential for additional checks and capabilities, esp. automatic inference of `skiprows`, but I think it's fine for now.

Thanks for improving this, it looks much more pythonic now 🐍 

Added a very minor change such that we can now glob subdirectories as I just changed to the following build layout
```
build
 |_ Debug
 |_ Release
```
